sonar-batch-protocol/src/main/java/org/sonar/batch/protocol/viewer/TextLineNumber.java

```diff
@@ -0,0 +1,460 @@
+    this.digitAlignment = digitAlignment > 1.0f ? 1.0f : digitAlignment < 0.0f ? -1.0f : digitAlignment;
+  }
+
+  /**
+   *  Gets the minimum display digits
+   *
+   *  @return the minimum display digits
+   */
+  public int getMinimumDisplayDigits() {
+    return minimumDisplayDigits;
+  }
+
+  /**
+   *  Specify the mimimum number of digits used to calculate the preferred
+   *  width of the component. Default is 3.
+   *
+   *  @param minimumDisplayDigits  the number digits used in the preferred
+   *                               width calculation
+   */
+  public void setMinimumDisplayDigits(int minimumDisplayDigits) {
+    this.minimumDisplayDigits = minimumDisplayDigits;
+    setPreferredWidth();
+  }
+
+  /**
+   *  Calculate the width needed to display the maximum line number
+   */
+  private void setPreferredWidth() {
+    Element root = component.getDocument().getDefaultRootElement();
+    int lines = root.getElementCount();
+    int digits = Math.max(String.valueOf(lines).length(), minimumDisplayDigits);
+
+    // Update sizes when number of digits in the line number changes
+
+    if (lastDigits != digits) {
+      lastDigits = digits;
+      FontMetrics fontMetrics = getFontMetrics(getFont());
+      int width = fontMetrics.charWidth('0') * digits;
+      Insets insets = getInsets();
+      int preferredWidth = insets.left + insets.right + width;
+
+      Dimension d = getPreferredSize();
+      d.setSize(preferredWidth, HEIGHT);
+      setPreferredSize(d);
+      setSize(d);
+    }
+  }
+
+  /**
+   *  Draw the line numbers
+   */
+  @Override
+  public void paintComponent(Graphics g) {
+    super.paintComponent(g);
+
+    // Determine the width of the space available to draw the line number
+
+    FontMetrics fontMetrics = component.getFontMetrics(component.getFont());
+    Insets insets = getInsets();
+    int availableWidth = getSize().width - insets.left - insets.right;
+
+    // Determine the rows to draw within the clipped bounds.
+
+    Rectangle clip = g.getClipBounds();
+    int rowStartOffset = component.viewToModel(new Point(0, clip.y));
+    int endOffset = component.viewToModel(new Point(0, clip.y + clip.height));
+
+    while (rowStartOffset <= endOffset) {
+      try {
+        if (isCurrentLine(rowStartOffset))
+          g.setColor(getCurrentLineForeground());
+        else
+          g.setColor(getForeground());
+
+        // Get the line number as a string and then determine the
+        // "X" and "Y" offsets for drawing the string.
+
+        String lineNumber = getTextLineNumber(rowStartOffset);
+        int stringWidth = fontMetrics.stringWidth(lineNumber);
+        int x = getOffsetX(availableWidth, stringWidth) + insets.left;
+        int y = getOffsetY(rowStartOffset, fontMetrics);
+        g.drawString(lineNumber, x, y);
+
+        // Move to the next row
+
+        rowStartOffset = Utilities.getRowEnd(component, rowStartOffset) + 1;
+      } catch (Exception e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)